src/django_rich/test.py

```diff
@@ -170,12 +177,86 @@ class RichTestRunner(DiscoverRunner.test_runner):
     resultclass = RichTextTestResult
 
 
+class RichTestSuite(unittest.TestSuite):
+    def run(
+        self, result: unittest.TestResult, debug: bool = False
+    ) -> unittest.TestResult:
+        topLevel = False
+        if getattr(result, "_testRunEntered", False) is False:
+            result._testRunEntered = topLevel = True  # type: ignore [attr-defined]
+
+        for index, test in enumerate(self):
+            if result.shouldStop:
+                break
+
+            if _isnotsuite(test):
+                self._tearDownPreviousClass(test, result)  # type: ignore [attr-defined]
+                self._handleModuleFixture(test, result)  # type: ignore [attr-defined]
+                self._handleClassSetUp(test, result)  # type: ignore [attr-defined]
+                result._previousTestClass = test.__class__  # type: ignore
+                # [attr-defined]
+                if getattr(test.__class__, "_classSetupFailed", False) or getattr(
+                    result, "_moduleSetUpFailed", False
+                ):
+                    continue
+            if not debug:
+                start_time = time.perf_counter()
+                test(result)
+                if isinstance(result, RichTextTestResult) and isinstance(
+                    test, TestCase
+                ):
+                    duration = time.perf_counter() - start_time
+                    result.collectedDurations.append((test, duration))
+
+            else:
+                test.debug()
+
+            # if self._cleanup: # Assume folk are not running python's regression tests.
+            # https://github.com/python/cpython/commit/8913a6c8
+            self._removeTestAtIndex(index)  # type: ignore [attr-defined]
+
+        if topLevel:
+            self._tearDownPreviousClass(None, result)  # type: ignore [attr-defined]
+            self._handleModuleTearDown(result)  # type: ignore [attr-defined]
+            result._testRunEntered = False  # type: ignore [attr-defined]
+        return result
+
+
 class RichRunner(DiscoverRunner):
     test_runner = RichTestRunner
+    test_suite = RichTestSuite
 
     def get_resultclass(self) -> type[RichTextTestResult] | None:
         if self.debug_sql:
             return RichDebugSQLTextTestResult
         elif django.VERSION >= (3, 0) and self.pdb:
             return RichPDBDebugResult
         return None
+
+    def suite_result(
+        self,
+        suite: unittest.TestSuite,
+        result: RichTextTestResult,
+        **kwargs: dict[Any, Any],
```

Comment:
actually the type here should be `Any` - see https://adamj.eu/tech/2021/05/11/python-type-hints-args-and-kwargs/
```suggestion
        **kwargs: Any,
```